src/Groups/gsets.jl

```diff
@@ -41,8 +41,9 @@ The fields are
     action_function::Function
     seeds
 
-    function GSetByElements(G::T, fun::Function, seeds; closed::Bool = false) where {T<:Union{GAPGroup, FinGenAbGroup}}
-        @assert !isempty(seeds)
+    function GSetByElements(G::T, fun::Function, seeds; closed::Bool = false, check::Bool = true) where {T<:Union{GAPGroup, FinGenAbGroup}}
+        @req !isempty(seeds) "seeds for g-set must be nonempty"
+        check && @req length(methods(fun, (typeof(first(seeds)), elem_type(T)))) != 0 "action function does not fit to seeds"
```

Comment:
```suggestion
        check && @req hasmethod(fun, (typeof(first(seeds)), elem_type(T))) "action function does not fit to seeds"
```
according to some discussion on the julia slack I read a few weeks ago, this is favorable to use